opm/simulators/linalg/ISTLSolverEbosFlexible.hpp

```diff
@@ -200,6 +256,46 @@ class ISTLSolverEbosFlexible
         }
     }
 
+    VectorType getTrueImpesWeights(const VectorType& b,const int pressureVarIndex)
+    {	    
+    	VectorType weights(b.size());
+    	BlockVector rhs(0.0);
+    	rhs[pressureVarIndex] = 1.0;
+    	int index = 0;
+    	ElementContext elemCtx(simulator_);
+    	const auto& vanguard = simulator_.vanguard();
+    	auto elemIt = vanguard.gridView().template begin</*codim=*/0>();
+    	const auto& elemEndIt = vanguard.gridView().template end</*codim=*/0>();
+    	for (; elemIt != elemEndIt; ++elemIt) {
+    	    const Element& elem = *elemIt;
+    	    elemCtx.updatePrimaryStencil(elem);
+    	    elemCtx.updatePrimaryIntensiveQuantities(/*timeIdx=*/0);
+	    const int numEq = MatrixType::block_type::rows;
+    	    Dune::FieldVector<Evaluation, numEq> storage;
+    	    unsigned threadId = ThreadManager::threadId();
+    	    simulator_.model().localLinearizer(threadId).localResidual().computeStorage(storage,elemCtx,/*spaceIdx=*/0, /*timeIdx=*/0);
+    	    Scalar extrusionFactor = elemCtx.intensiveQuantities(0, /*timeIdx=*/0).extrusionFactor();
+    	    Scalar scvVolume = elemCtx.stencil(/*timeIdx=*/0).subControlVolume(0).volume() * extrusionFactor;
+    	    Scalar storage_scale = scvVolume / elemCtx.simulator().timeStepSize();
+    	    MatrixBlockType block;
+    	    double pressure_scale = 50e5;
+    	    for (int ii = 0; ii < numEq; ++ii) {
+    		for (int jj = 0; jj < numEq; ++jj) {
+    		    block[ii][jj] = storage[ii].derivative(jj)/storage_scale;
+    		    if (jj == pressureVarIndex) {
+    			block[ii][jj] *= pressure_scale;
+    		    }
+    		}
+    	    }
+    	    BlockVector bweights;
+    	    MatrixBlockType block_transpose = Opm::transposeDenseMatrix(block);
+    	    block_transpose.solve(bweights, rhs);
+    	    bweights /= 1000.0; // given normal densities this scales weights to about 1.
+    	    weights[index] = bweights;
+    	    ++index;
+    	}
+    	return weights;
```

Comment:
Does the computation in the above loop need values for cell neighbors? If that is the case, then we are missing communication when running parallel.

Reply:
The weights a calculated only based on the diagonal block quasiimpes or using the blockmatrix generated from the accumulation term trueimpes